pkg/snapshot/restorer/restorer.go

```diff
@@ -155,21 +171,21 @@ func (r *Restorer) restoreFromBaseSnapshot(ro brtypes.RestoreOptions) error {
 		return err
 	}
 
-	memberDir := filepath.Join(ro.Config.RestoreDataDir, "member")
+	memberDir := filepath.Join(ro.Config.DataDir, "member")
 	if _, err := os.Stat(memberDir); err == nil {
 		return fmt.Errorf("member directory in data directory(%q) exists", memberDir)
 	}
 
 	walDir := filepath.Join(memberDir, "wal")
-	snapdir := filepath.Join(memberDir, "snap")
-	if err = r.makeDB(snapdir, ro.BaseSnapshot, len(cl.Members()), ro.Config.SkipHashCheck); err != nil {
+	snapDir := filepath.Join(memberDir, "snap")
+	if err = r.makeDB(snapDir, ro.BaseSnapshot, len(cl.Members()), ro.Config.SkipHashCheck); err != nil {
 		return err
 	}
-	return makeWALAndSnap(r.zapLogger, walDir, snapdir, cl, ro.Config.Name)
+	return makeWALAndSnap(r.zapLogger, walDir, snapDir, cl, ro.Config.Name)
 }
 
 // makeDB copies the database snapshot to the snapshot directory.
-func (r *Restorer) makeDB(snapdir string, snap *brtypes.Snapshot, commit int, skipHashCheck bool) error {
+func (r *Restorer) makeDB(snapDir string, snap *brtypes.Snapshot, commit int, skipHashCheck bool) error {
 	rc, err := r.store.Fetch(*snap)
```

Comment:
Ok, I'll change the variable names to make it easier to differentiate between compressed and decompressed files and RCs.

Reply:
Actually on second thought, the generic name `rc` is used because etcd-backup-restore does not know whether the snapshot from the bucket is compressed or not. We cannot assume that it was a compressed snapshot, because users can disable snapshot compression when running etcdbrctl. And you can also see on line 201 that the same variable `rc` is being assigned the readCloser for the decompressed snapshot, so that we don't need to add any extra code after this to choose between two separate variables for compressed and decompressed RCs.

I would vote to keep the name as is.

Reply:
ok lets keep it the way it is now.